src/lib.rs

```diff
@@ -166,6 +179,22 @@ impl LightningNode {
         let _graph = persister.read_graph();
 
         // Step 12. Initialize the PeerManager
+        let mut ephemeral_bytes = [0; 32];
+        OsRng.try_fill_bytes(&mut ephemeral_bytes).map_err(|e| {
```

Comment:
Should we move out the crypto stuff into its own module where it can be reused?

Reply:
I was starting to do that and then realised that it wouldn't be that reusable due to the returned error. To make it reusable, we would still have to `map_err()` in both places this is used, and we would end up with almost the same repeated code.

Reply:
I will have a look into it.